pkg/tests/tasks/traffic/egress/access_external_services_test.go

```diff
@@ -15,102 +15,150 @@
-		command := `curl -sSI https://www.redhat.com/en | grep  "HTTP/"`
-		msg, err := util.PodExec("bookinfo", sleepPod, "sleep", command, false)
-		util.Inspect(err, "Failed to get response", "", t)
-		if strings.Contains(msg, "200") {
-			log.Log.Infof("Success. Get https://www.redhat.com/en response: %s", msg)
-		} else {
-			log.Log.Infof("Error response: %s", msg)
-			t.Errorf("Error response: %s", msg)
-		}
-	})
+	test.NewTest(t).Id("T11").Groups(test.Full, test.InterOp).Run(func(t test.TestHelper) {
+		hack.DisableLogrusForThisTest(t)
 
-	t.Run("TrafficManagement_egress_controlled_access_to_external_httpbin_services", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-
-		log.Log.Info("Skip update global.outboundTrafficPolicy.mode")
-		log.Log.Info("Create a ServiceEntry to external httpbin")
-		util.KubeApplyContents("bookinfo", httpbinextServiceEntry)
-		time.Sleep(time.Duration(10) * time.Second)
-		command := `curl -sS http://httpbin.org/headers`
-		msg, err := util.PodExec("bookinfo", sleepPod, "sleep", command, false)
-		if err != nil {
-			log.Log.Infof("Error response: %s", msg)
-			t.Errorf("Error response: %s", msg)
-		} else {
-			log.Log.Infof("Success. Get http://httpbin.org/headers response:\n%s", msg)
-		}
-	})
+		ns := "bookinfo"
+		meshNamespace := env.GetDefaultMeshNamespace()
+		smcpName := env.GetDefaultSMCPName()
+		outboundDefaultPatch := `[{"op": "remove", "path": "/spec/proxy/networking/trafficControl/outbound/policy"}]`
+		t.Cleanup(func() {
+			app.Uninstall(t, app.Sleep(ns))
+			oc.Patch(
+				t,
+				meshNamespace,
+				"smcp",
+				smcpName,
+				"json",
+				outboundDefaultPatch,
+			)
+		})
 
-	t.Run("TrafficManagement_egress_access_to_external_https_redhat", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-
-		log.Log.Info("Create a ServiceEntry to external https://www.redhat.com/en")
-		util.KubeApplyContents("bookinfo", redhatextServiceEntry)
-		time.Sleep(time.Duration(10) * time.Second)
-		command := `curl -sSI https://www.redhat.com/en | grep  "HTTP/"`
-		msg, err := util.PodExec("bookinfo", sleepPod, "sleep", command, false)
-		util.Inspect(err, "Failed to get response", "", t)
-		if strings.Contains(msg, "200") {
-			log.Log.Infof("Success. Get https://www.redhat.com/en response: %s", msg)
-		} else {
-			log.Log.Infof("Error response: %s", msg)
-			t.Errorf("Error response: %s", msg)
-		}
-	})
+		t.Log("This test validates accesses to external services")
+
+		t.LogStepf("Install sleep into %s", ns)
+		app.InstallAndWaitReady(t, app.Sleep(ns))
+
+		t.LogStep("Make request to www.redhat.com from sleep")
+		execInSleepPod(
+			t,
+			ns,
+			buildGetRequestCmd("https://www.redhat.com/en"),
+			assert.OutputContains(
+				"200",
+				"Got expected 200 ok from www.redhat.com",
+				"Expect 200 ok from www.redhat.com, but got a different HTTP code",
+			),
+		)
+
+		outboundRegistryOnlyPatch := `[{"op": "add", "path": "/spec/proxy/networking/trafficControl/outbound/policy", "value": "REGISTRY_ONLY"}]`
```

Comment:
inline

Reply:
Also, this failed on my machine:
```
access_external_services_test.go:69: FATAL: Command failed: "oc patch -n istio-system smcp/basic --type json -p '[{\"op\": \"add\", \"path\": \"/spec/proxy/networking/trafficControl/outbound/policy\", \"value\": \"REGISTRY_ONLY\"}]'"
        Error: command failed: The request is invalid: the server rejected our request due to an error in our request
```

Reply:
To use patch, it requires the all nodes on the path to exist. Setting `/A/B/C` will require `A` and `B` to be there.